tests/unit/set_tags/test_set_bucket_apply_tags.py

```diff
@@ -0,0 +1,49 @@
+import unittest
+from unittest.mock import MagicMock
+
+from botocore.stub import Stubber
+
+from set_tags import set_bucket_tags
+from set_tags import utils
+
+
+class TestSetInstanceTagsHandler(unittest.TestCase):
+
+  TEST_BUCKET_NAME = 'my-bucket'
+  TEST_TAGS = [{
+    'Key': 'foo',
+    'Value': 'bar'
+  }]
+
+  def test_happy_path(self):
+    s3 = utils.get_s3_client()
+    with Stubber(s3) as stubber:
+      stubber.add_response(
+        method='put_bucket_tagging',
+        expected_params={
+          'Bucket': self.TEST_BUCKET_NAME,
+          'Tagging': {
+            'TagSet': self.TEST_TAGS
+          }
+        },
+        service_response={
+          'ResponseMetadata': {
+            'RequestId': '12345',
+            'HostId': 'etc',
+            'HTTPStatusCode': 204,
+            'HTTPHeaders': {}
+          }})
+      utils.get_s3_client = MagicMock(return_value=s3)
+      result = set_bucket_tags.apply_tags(self.TEST_BUCKET_NAME, self.TEST_TAGS)
+
+
+  def test_client_error(self):
+    s3 = utils.get_s3_client()
+    with Stubber(s3) as stubber, self.assertRaises(Exception):
+      stubber.add_client_error(
+        method='put_bucket_tagging',
+        service_error_code='NoSuchBucket',
+        service_message='The specified bucket does not exist',
+        http_status_code=404)
+      utils.get_s3_client = MagicMock(return_value=s3)
+      result = set_bucket_tags.apply_tags(self.TEST_BUCKET_NAME, self.TEST_TAGS)
```

Comment:
If the philosophy in this custom resource is to skip error handling and instead rely on crhelper to surface everything, then what are you testing here? If apply tags were catching the exception then rethrowing a more intelligible error, this would make more sense. Instead, this verifies that Stubber is throwing an exception when you tell it to do so.

Reply:
yup, agreed.  test has been removed.